### PR TITLE
Added documentation for classification

### PR DESCRIPTION
**Pull request by QuincySharp**

### CODE REVIEW

File: src/classification/__main__.py [hunk 1]
```diff
@@ -3,8 +3,9 @@
 import os
 import re
 import sys
-from multiprocessing import Pool
 
+from typing import Tuple
+from multiprocessing import Pool
```
Comment (Areedtz) [inline src/classification/__main__.py]: Multiprocessing is a base python module and should be at the top :)

File: src/classification/__main__.py [hunk 2]
```diff
@@ -15,7 +16,23 @@
 from extractor.low_level_data_extractor import make_low_level_data_file
 
 
-def process_data_and_extract_profiles(song_id, song_file, song_output_file):
+def process_data_and_extract_profiles(song_id: str, song_file: str, song_output_file: str) -> Tuple[str, Tuple, Tuple, Tuple, Tuple, Tuple, Tuple]:
```
Comment (EmilHenriksen) [inline src/classification/__main__.py]: Length pep8

File: src/classification/classifier/profile_data_extractor.py [hunk 3]
```diff
@@ -13,7 +14,20 @@
 from utilities.filehandler.handle_path import get_absolute_path
 
 
-def get_classifier_data(data_file_name):
+def get_classifier_data(data_file_name: str) -> Tuple[Tuple, Tuple, Tuple, Tuple, Tuple, Tuple]:
```
Comment (EmilHenriksen) [inline src/classification/classifier/profile_data_extractor.py]: maybe length pep8?

File: src/classification/classifier/profile_data_extractor.py [hunk 4]
```diff
@@ -54,8 +68,7 @@ def get_classifier_data(data_file_name):
 
     # list for beautifying code
     t = [(timbre, timbre_probability), (mood_relaxed, mood_relaxed_probability),
-         (mood_party, mood_party_probability), (mood_aggressive,
-                                                mood_aggressive_probability),
+         (mood_party, mood_party_probability), (mood_aggressive, mood_aggressive_probability),
```
Comment (EmilHenriksen) [inline src/classification/classifier/profile_data_extractor.py]: maybe length pep8

File: src/classification/segmented_audio_analysis.py [hunk 5]
```diff
@@ -25,7 +26,21 @@
 from utilities.get_song_id import get_song_id
 
 
-def process_data_and_extract_profiles(segment_id, song_file):
+def process_data_and_extract_profiles(segment_id: int, song_file) -> Tuple[int, Tuple, Tuple, Tuple, Tuple, Tuple, Tuple, Tuple]:
```
Comment (EmilHenriksen) [inline src/classification/segmented_audio_analysis.py]: length pep8

File: src/classification/segmented_audio_analysis.py [hunk 6]
```diff
@@ -7,8 +7,9 @@
 import subprocess
 import sys
 import tempfile
-from multiprocessing import Pool
 
+from multiprocessing import Pool
```
Comment (Areedtz) [inline src/classification/segmented_audio_analysis.py]: Multiprocessing is a base python module and should be at the top.

File: src/classification/segmented_audio_analysis.py [hunk 7]
```diff
@@ -25,7 +26,21 @@
...
+        id of the segment of the audiofile corresponds to
+    song_file
+        the audio from a segmented song
+
+    Returns
+    -------
+    Tuple[int, Tuple, Tuple, Tuple, Tuple, Tuple, Tuple, Tuple]
+        A tuple of the segment_id and tuples describing BPM, timbre and all moods
+    """
     # creating the temporary files
```
Comment (Areedtz) [inline src/classification/segmented_audio_analysis.py]: You should have a space between the docstring and the code

File: src/classification/__main__.py [hunk 8]
```diff
@@ -15,7 +16,23 @@
...
+        the filepath of the given song
+    song_output_file
+        the output file for the lowlevel datafile
+
+    Returns
+    -------
+    Tuple[str, Tuple, Tuple, Tuple, Tuple, Tuple, Tuple]
+        A tuple of the song_id and tuples describing all moods and their probability
+    """
     make_low_level_data_file(song_file, song_output_file)
```
Comment (Areedtz) [inline src/classification/__main__.py]: You should have a space between the docstring and the code

File: src/classification/__main__.py (hunk 2 above)
Comment (Areedtz) [inline src/classification/__main__.py]: Should probably look into the line length

File: src/classification/segmented_audio_analysis.py (hunk 5 above)
Comment (Areedtz) [inline src/classification/segmented_audio_analysis.py]: Should probably look into the line length.

File: src/classification/segmented_audio_analysis.py [hunk 9]
```diff
@@ -49,7 +64,21 @@ def process_data_and_extract_profiles(segment_id, song_file):
...
+        path to the directory
+    song_file
+        path to the given song
+
+    Returns
+    -------
+    Tuple[str, str, list]
+        A tuple of the song_id, directory path and a list with the segments and their ids
+    """
     dirname = os.path.abspath(os.path.dirname(filename))
```
Comment (Areedtz) [inline src/classification/segmented_audio_analysis.py]: You should have a line between the docstring and the code.

File: src/classification/extractor/low_level_data_extractor.py [hunk 10]
```diff
@@ -7,7 +7,17 @@
...
+    and outputs in the given outputfile
+
+    Parameters
+    ----------
+    filename
+        single song file path
+    output_file_path
+        output path of the lowlevel datafile
+    """
     extractor_path = path_handler.get_absolute_path("utilities/ressources"
```
Comment (Areedtz) [inline src/classification/extractor/low_level_data_extractor.py]: You should have a line between the docstring and the code.

**Pull request merged**